app/models/league.rb

```diff
@@ -44,6 +48,23 @@ def calculate_round_robin
     save
   end
 
+  def calculate_two_halfs
+    pairings_per_day = round_robin_pairings teams.to_a
+    pairings_per_day += round_robin_pairings teams.to_a
+    pairings_per_day.each_with_index do |day, gameday|
+      day.each do |pairing|
+        # Creating a match for every pairing if one of the teams is nil (which happens if there is an odd number of teams)
+        # the other team will have to wait for this day
+        if gameday < teams.size
+          matches << Match.new(team_home: pairing[0], team_away: pairing[1], gameday: gameday + 1) unless pairing[0].nil? || pairing[1].nil?
```

Comment:
We can chain a map and select call `round_robin_pairings` method and remove the nil pairings with it. That way we don't have to catch the edge case here anymore.